src/index.js

```diff
@@ -126,6 +121,40 @@ async function bundleMDX(
     },
   }
 
+  /** @type import('esbuild').Plugin */
+  const xdmPlugin = {
+    name: 'xdm',
+    setup(build) {
+      build.onLoad({filter: /\.mdx$/}, async ({path: filePath, pluginData}) => {
```

Comment:
If I'm not mistaken, by doing this, we will not be able to handle files with extensions other than `.mdx`
The original xdm plugin has a few options to control extensions, and by default, ends up using this as a filter:
`/\.(md|markdown|mdown|mkdn|mkd|mdwn|mkdown|ron|mdx)$/`